lib/Myriad/Bootstrap.pm

```diff
@@ -171,50 +232,41 @@ sub boot {
                 die "excessive module loading detected: $loaded_modules" if $loaded_modules;
             }
 
-            my $stop = 0;
             local $SIG{HUP} = sub {
                 say "$$ - HUP detected in parent";
                 kill 3, $pid;
-    #           $stop = 1;
             };
 
-            # Build up any output from the child process
-            my $input = '';
+            print $child_pipe "Parent active\r\n";
 
-            my $rin = my $win = '';
-            vec($rin, fileno($child_pipe), 1) = 1;
-            my $ein = $rin | $win;
             ACTIVE:
-            while(!$stop) {
-                say "$$ Parent loop cycle";
-                die $! unless defined(my $nfound = select my $rout = $rin, my $wout = $win, my $eout = $ein, 5);
-                if($nfound) {
-    #               say "$$ Child has something to report ($nfound): $rout, $wout, $eout";
-                    my $rslt = sysread $child_pipe, my $buf, 4096;
-                    last ACTIVE unless $rslt;
-                    $input .= $buf;
-                    while($input =~ s/^[\r\n]*(.*)[\r\n]+//) {
-                        say "Received from child: $1";
+            while (1) {
+
+                last ACTIVE unless check_messages_in_pipe($inotify_child_pipe, sub {
+                    say "$$ - File has been detected reloading..";
+                    kill 3, $pid;
```

Comment:
I take it here you want to signal the out-of-date process to stop to then start a new instance with the up-to-date modules loaded. But the kid might have resources locked like port numbers. Also, it is not instantly dead as soon as the bullet leaves the barrel. So, the successor might not be able to start up. Maybe better to wait for it to die here.

But I may be missing something.